tests/unit_tests/client/address_book/storage.cc

```diff
@@ -0,0 +1,192 @@
+ *    materials provided with the distribution.                                               //
+ *                                                                                            //
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be       //
+ *    used to endorse or promote products derived from this software without specific         //
+ *    prior written permission.                                                               //
+ *                                                                                            //
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY        //
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF    //
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL     //
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,       //
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,               //
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS    //
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,          //
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF    //
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.               //
+ */
+
+#define BOOST_TEST_DYN_LINK
+
+#include <boost/test/unit_test.hpp>
+
+#include <iostream>
+
+#include "client/address_book/impl.h"
+#include "client/address_book/storage.h"
+
+#include "core/router/context.h"
+
+namespace client = kovri::client;
+namespace core = kovri::core;
+
+struct AddressBookStorageFixture
+{
+  /// @alias Subscription
+  /// @brief Subscription alias for readability and convenience
+  /// @details Intended for user-convenience and readability
+  using SubscriptionType = client::AddressBook::SubscriptionType;
+
+  AddressBookStorageFixture() : m_Storage(nullptr)
+  {
+    // Setup temporary data directory
+    core::context.SetCustomDataDir(temp_path.string());
+
+    // Ensure Client and AddressBook paths exist
+    core::EnsurePath(core::GetPath(core::Path::Client));
+
+    core::EnsurePath(core::GetPath(core::Path::AddressBook));
+
+    // Setup storage
+    m_Storage = GetStorageInstance();
+  }
+
+  ~AddressBookStorageFixture()
+  {
+    RemoveFiles();
+  }
+
+  /// @brief Remove files created by AddressBookStorage
+  void RemoveFiles()
+  {
+    try
+      {
+        boost::filesystem::directory_iterator end_itr;
+        for (boost::filesystem::directory_iterator dir_itr(
+                 core::GetPath(core::Path::AddressBook));
+             dir_itr != end_itr;
+             ++dir_itr)
+          {
+            auto const temp_file = dir_itr->path().filename();
+
+            // Only remove files created by AddressBookStorage
```

Comment:
Remove everything. We shouldn't have lingering files after running a test.

Reply:
Ok, this was left over from creating random files with `boost::unique_path`. I'll remove all files.

Reply:
So, when I woke up and PRs were running through my head as I laid in bed, I realized that what I meant to say was *when the unit-test was finished*. Because you `RemoveFiles()` in the fixture dtor, it will remove X for every test-case. I don't want to get in your way if you want to play with files throughout the test, what I meant was that, when the test is finished, remove everything :grin: